hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/handlers/TopicHandler.java

```diff
@@ -55,10 +54,14 @@ private void onTopicPresent(HttpServerExchange exchange, String messageId, Consu
         try {
             Optional<CachedTopic> topic = topicsCache.getTopic(qualifiedTopicName);
             if (topic.isPresent()) {
-                consumer.accept(topic.get());
-                return;
+                if (!topicsCache.isBlacklisted(topic.get().getQualifiedName())) {
```

Comment:
why call get twice in the main loop., also -> we have qualifiedTopicName here, no need to construct it once again (pointing it out, because this is probably the hottest code in whole frontend module)